### PR TITLE
ci: add pr title lint

### PR DESCRIPTION
**Pull request by xuthus5**
Help unify and standardize pr titles.

### CODE REVIEW

**Review by shoothzj — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,6 +20,15 @@ env:
   GOSUMDB: off
 
 jobs:
+  PrTitleLint:
+    name: pr title lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
```
Comment (shoothzj) [inline .github/workflows/ci.yml:29]: fetch-depth might be the default value?

Comment (xuthus5) [reply .github/workflows/ci.yml:29]: The full-depth may be too large.

Comment (shoothzj) [reply .github/workflows/ci.yml:29]: @xuthus5 
**Only a single commit is fetched by default, for the ref/SHA that triggered the workflow.**

So I think we can omit this config

Comment (xuthus5) [reply .github/workflows/ci.yml:29]: OK,I found a project to verify whether context.payload requires git history. Maybe this step is redundant.

**Review by shoothzj — APPROVED** [review]

**Pull request merged**